### ISSUE TITLE
Change from Newtonsoft to System.Text.Json

### ISSUE THREAD

**Issue by Chrislock1** [dotnetrdf/dotnetrdf#538]
To support new stacks, Newtonsoft should be replaced with System.Text.Json.

**Comment by kal** [comment]
Can you elaborate on this please. Is there some clash that is happening when you try to use DNR in a project or is this just a case of suggesting that we change from Newtonsoft to the MS JSON parser?

**Comment by zotanmew** [comment]
I can only speculate what the original issue author meant here, but I'm currently running into an issue where I'd really like to use the System.Text.Json response serializer in an ASP.NET application that also uses dotnetrdf, however the object types are incompatible and converting between the two is [really annoying](https://stackoverflow.com/questions/65386083/how-can-i-serialize-a-newtonsoft-jtoken-to-json-using-system-text-json).

**Comment by zotanmew** [comment]
I'm going to have a look at how feasible this is to do. If it's not too much work I'll implement this as well :)

**Comment by kal** [comment]
Yeah, its quite deeply embedded unfortunately and it also leaks into some of our public APIs (e.g. the public methods on `JsonLdProcessor`) which means that ideally any transition would deprecate the APIs that use Newtonsoft.Json but leave them intact for at least one full major release (i.e. 4.x would have to have both APIs with the Newtonsoft.JSON ones deprecated then 5.x can remove them). However, I realise that this might make transition entirely impossible (or at least too much work to be considered to be possible!), in which case we would have to just make it a breaking change...there is already precedence for that in the transition from 2.x to 3.x, but I'd like to minimise the impact of major release updates as much as possible.

**Comment by zotanmew** [comment]
Understandable! I've worked around this for now using a custom controller formatter that defaults to system.text.json and uses newtonsoft.json if a specific attribute is set on the route/controller, but I'll likely still want to resolve this long term, either via upstreaming the migration, making a custom fork for the meantime, or writing my own RDF/JSON-LD library. But with the workaround the incentive to do this soon has decreased significantly, so I may just wait for someone else to pick this up :)

**Comment by CFlatWouldbeMinor** [comment]
I would also like to contribute to this topic, Again also looking into performance (Sorry again related to performance issue #300 , besides performance on certain use cases I really like this project) I think reading json could be improved. If I compare the load speed of the same project between nquads and JSONLD it is about a factor 300x. Majority of the time (In my case 83%) is really spent in reading json..
There would also be some AI from MSFT be available.
As System.Text.Json has evolved quite significantly it might be also an option to use the new concept only on newer .net (10+)versions and not .net standard. 2.0. At a certain point it might be useful to add a new package like the async to profit from the new features. Keep the old version e.g 4.0 as is based on the old parse and add a new package as addon as seperate nuget as mentioned @kal in the above post

**Comment by wont-work** [comment]
i have rewritten most of the json-ld algorithms (expansion, compaction, flattening) with s.t.j in https://iceshrimp.dev/iceshrimp/LinkedData.NET. the license is likely a bit too restrictive for a lot of dotnetrdf users (it's EUPL) and there aren't any stable releases (maybe ever?), but at least it proves the perf benefits from switching to s.t.j are real: https://it.w.on-t.work/ldnet2025.html

still have not managed to wire it up into the project i wrote it for, so maybe it falls over in the real world, but the tests for the features i specified pass.

**Comment by kal** [comment]
I've done a port of the existing code from Newtonsoft.Json to System.Text.Json.Nodes on #891 

Because of the way that Newtonsoft.Json types leaked into our APIs, this will result in a breaking API change. I looked for a way to work around it and try to marshall between the libraries, but it turned out not to be possible so I have just gone for a breaking change rather than keep two parallel implementations in place. As noted on that PR there may be some additional optimisation to do around how often nodes are cloned in the processing algorithms. However, in terms of correctness, all of the W3C test suite that passed with the Newtonsoft implementation now pass with the System.Text.Json implementation.
